app/services/bookings/gitis/subject_fetcher.rb

```diff
@@ -0,0 +1,26 @@
+module Bookings
+  module Gitis
+    class SubjectFetcher
+      def api_subject_id_from_gitis_value(value)
+        return nil if value.blank?
+
+        teaching_subjects.find { |s| s.value == value }&.id
+      end
+
+      def api_subject_from_gitis_uuid(uuid)
+        return nil if uuid.blank?
+
+        teaching_subjects.find { |s| s.id == uuid }&.value
+      end
+
+    private
+
+      def teaching_subjects
+        @teaching_subjects ||= begin
```

Comment:
It may be worth having this as a module or class method so the memoization is broader?